CMakeLists.txt

```diff
@@ -114,15 +114,30 @@ if(USE_RTOS)
     set(THREADX_ARCH "cortex_m4")
     set(THREADX_TOOLCHAIN "gnu")
     add_subdirectory(libs/threadx)
-    target_sources(${PROJECT_NAME} PRIVATE src/core/rtos/f4xx/tx_initialize_low_level.s)
+
+    set(THREADX_DIR ${CMAKE_SOURCE_DIR}/libs/threadx)
+
+    target_include_directories(${PROJECT_NAME} PRIVATE
```

Comment:
This is to avoid the whole thing that happened with CANopen where we had to modify every CMAKE file in the library to change how it was exposed. CMAKE is dumb and it does not like submodules. Also this way updates to threadX are just a pullup and fixing any breaking changes.

Reply:
Yeah, tbh I don't know why this works I kinda maybe a little bit just asked ChatGPT and verified that it worked.